activerecord/lib/active_record/associations/collection_proxy.rb

```diff
@@ -715,21 +715,24 @@ def distinct
       end
       alias uniq distinct
 
-      # Count all records using SQL.
+      # Count all records.
       #
       #   class Person < ActiveRecord::Base
       #     has_many :pets
       #   end
       #
       #   person.pets.count # => 3
+      #   # This will perform the count using SQL.
       #   person.pets
       #   # => [
       #   #       #<Pet id: 1, name: "Fancy-Fancy", person_id: 1>,
       #   #       #<Pet id: 2, name: "Spook", person_id: 1>,
       #   #       #<Pet id: 3, name: "Choo-Choo", person_id: 1>
       #   #    ]
-      def count(column_name = nil)
-        @association.count(column_name)
+      #   person.pets.count { |pet| pet.name.include?('-') } # => 2
+      #   # Passing a block will select all pets in SQL and perform the count using Ruby.
```

Comment:
Interesting. Haven't seen that format used before to be honest. If we're going to follow file precedent, we should make it more "precedential": `# executes something like <SQL query>` and newlines.